{{cookiecutter.directory_name}}/README.dev.md

```diff
@@ -138,35 +138,23 @@ This section describes how to make a release in 3 parts:
 
 ### (2/3) PyPI
 
-In a new terminal, without an activated virtual environment or an env directory:
+In a new terminal:
 
 ```shell
-# prepare a new directory
+# OPTIONAL: prepare a new directory with fresh git clone to ensure the release
+# has the state of origin/main branch
 cd $(mktemp -d {{ cookiecutter.package_name }}.XXXXXX)
-
-# fresh git clone ensures the release has the state of origin/main branch
 git clone {{ cookiecutter.repository }} .
 
-# prepare a clean virtual environment and activate it
-python -m venv env
-source env/bin/activate
-
-# make sure to have a recent version of pip and setuptools
-python -m pip install --upgrade pip setuptools
-
-# install runtime dependencies and publishing dependencies
-python -m pip install --no-cache-dir .
-python -m pip install --no-cache-dir .[publishing]
-
-# clean up any previously generated artefacts
-rm -rf {{ cookiecutter.package_name }}.egg-info
-rm -rf dist
+# make sure to have a recent version of pip and the publishing dependencies
+python -m pip install --upgrade pip
+python -m pip install .[publishing]
 
 # create the source distribution and the wheel
-python setup.py sdist bdist_wheel
+python -m build
 
 # upload to test pypi instance (requires credentials)
-twine upload --repository-url https://test.pypi.org/legacy/ dist/*
+python -m twine upload --repository testpypi dist/*
```

Comment:
Out of curiosity: why is it better to use `python -m twine` than just `twine`? Or `python -m pip` instead of just `pip`?

Reply:
Because if you're in a virtualenv that doesn't have `pip` installed, but you do have a system-wide pip (e.g. the one from `apt install python-pip`) then you end up running a version of pip that doesn't match the version of Python in your virtualenv, and all hell breaks loose.

`python -m pip` is guaranteed to be whichever `pip` matches whichever `python` you're calling, so you avoid that problem.

Reply:
Perhaps a small tip to add to the Python Guide chapter. Although also maybe not so eScience specific that it makes sense there.

Reply:
Thanks for explaining! I'm not sure if I find the little bit of extra safety worth all the extra typing though. In most cases you would just get a permission denied error if you accidentally run the sytem-wide pip and the system version of twine will probably upload your package just fine.

Reply:
Sure, I also most of the time don't type the full thing. This is the template, though, and here it doesn't hurt to have to type the safest version once and get the benefits many times :) I have run into errors with these kinds of things several times. On CI, for instance, you also sometimes get these kinds of fun issues.